src/main/scala/cromwell/engine/workflow/WorkflowActor.scala

```diff
@@ -248,29 +251,36 @@ case class WorkflowActor(workflow: WorkflowDescriptor, backend: Backend)
       transition(WorkflowRunning) using data.copy(startMode = Option(startMode))
   }
 
-  def callCompletedWhileRunning(callKey: OutputKey, outputs: CallOutputs, returnCode: Int, message: TerminalCallMessage): State = {
-    val nextState = awaitCallComplete(callKey, outputs, returnCode) match {
-      case Success(_) =>
-        if (isWorkflowDone) transition(WorkflowSucceeded) else startRunnableCalls()
-      case Failure(e) =>
-        logger.error(e.getMessage, e)
+  private def handleCallCompleted(callKey: OutputKey, outputs: CallOutputs, returnCode: Int, message: TerminalCallMessage): State = {
+    // Don't close over sender().
```

Comment:
Basically we want the acknowledgment to go to the sender of the message currently being processed, but putting `sender()` in the forcomp below would cause it to be evaluated in the future when that method might return a different value.

http://helenaedelson.com/?p=879